trunk/SUAVE/Methods/Aerodynamics/Supersonic_Zero/Drag/parasite_drag_wing.py

```diff
@@ -0,0 +1,272 @@
+
+    Inputs:
+    settings.wing_parasite_drag_form_factor      [Unitless]
+    state.conditions.freestream.
+      mach_number                                [Unitless]
+      temperature                                [K]
+      reynolds_number                            [Unitless]
+    geometry.
+      areas.reference                            [m^2]
+      chords.mean_aerodynamic                    [m]
+      thickness_to_chord                         [Unitless]
+      sweeps.quarter_chord                       [radians]
+      aspect_ratio                               [Unitless]
+      spans.projected                            [m]
+      areas.exposed                              [m^2]
+      areas.affected                             [m^2]
+      areas.wetted                               [m^2]
+      transition_x_upper                         [Unitless]
+      transition_x_lower                         [Unitless]
+      
+      
+    Outputs:
+    wing_parasite_drag                           [Unitless]
+
+    Properties Used:
+    N/A
+    """
+    
+    # unpack inputs
+    C = settings.wing_parasite_drag_form_factor
+    freestream = state.conditions.freestream
+    
+    # conditions
+    Mc  = freestream.mach_number
+    Tc  = freestream.temperature    
+    re  = freestream.reynolds_number     
+    
+    wing = geometry
+    wing_parasite_drag = 0.0
+    
+    # Unpack wing
+    exposed_root_chord_offset = wing.exposed_root_chord_offset
+    symm                      = wing.symmetric
+    semispan                  = wing.spans.projected*0.5 * (2 - symm)
+    t_c_w                     = wing.thickness_to_chord
+    Sref                      = wing.areas.reference
+    num_segments              = len(wing.Segments.keys())     
+    
+    # if wing has segments, compute and sum parasite drag of each segment
+    
+    if num_segments>0:        
+        total_wetted_area            = 0
+        total_segment_parasite_drag  = 0 
+        total_segment_k_w            = 0 
+        total_segment_cf_w_u         = 0
+        total_segment_cf_w_l         = 0 
+        total_segment_k_comp_u       = 0
+        total_segment_k_comp_l       = 0
+        total_k_reyn_u               = 0  
+        total_k_reyn_l               = 0    
+        root_chord                   = wing.chords.root  
+        
+        for i_segs in range(num_segments):
+            if i_segs == num_segments-1:
+                continue 
+            else:  
+                span_seg  = semispan*(wing.Segments[i_segs+1].percent_span_location - wing.Segments[i_segs].percent_span_location )            
+                sweep_seg = wing.Segments[i_segs].sweeps.quarter_chord    
+                xtu       = wing.transition_x_upper
+                xtl       = wing.transition_x_lower      
+                
+                if i_segs == 0:
+                    chord_root    = root_chord*wing.Segments[i_segs].root_chord_percent
```

Comment:
many of these calcs between 0 and else are the same. So you should just pull them out

Reply:
I've deleted this file since I realized it could be combined easily with fidelity zero, but these issues still exist in the fidelity zero method.